File: backend-python/utils/audio_utils.py
"""Audio utility functions for backend processing"""
import base64
from typing import Any


def decode_base64(base64_str: str) -> bytes:
    """Converts a base64 string to bytes."""
    return base64.b64decode(base64_str)


def encode_base64(data: bytes) -> str:
    """Converts bytes to base64 string."""
    return base64.b64encode(data).decode('utf-8')


def validate_audio_data(data: Any) -> bool:
    """Validates audio data format"""
    if not data or not isinstance(data, dict):
        return False
    if 'data' not in data or not isinstance(data['data'], str):
        return False
    if 'mimeType' not in data or 'audio' not in data['mimeType']:
        return False
    return True

File: backend/src/utils/audioUtils.ts
/**
 * Audio utility functions for backend processing
 */

/**
 * Converts a base64 string to a Uint8Array.
 */
export function decodeBase64(base64: string): Uint8Array {
  const binaryString = Buffer.from(base64, 'base64').toString('binary');
  const len = binaryString.length;
  const bytes = new Uint8Array(len);
  for (let i = 0; i < len; i++) {
    bytes[i] = binaryString.charCodeAt(i);
  }
  return bytes;
}

/**
 * Converts a Uint8Array to base64 string.
 */
export function encodeBase64(bytes: Uint8Array): string {
  return Buffer.from(bytes).toString('base64');
}

/**
 * Validates audio data format
 */
export function validateAudioData(data: any): boolean {
  if (!data || typeof data !== 'object') return false;
  if (!data.data || typeof data.data !== 'string') return false;
  if (!data.mimeType || !data.mimeType.includes('audio')) return false;
  return true;
}

